Assets/_BForBoss/_Weapons/Scripts/Weapons/MeleeWeaponBehaviour.cs

```diff
@@ -0,0 +1,101 @@
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+namespace Perigon.Weapons
+{
+    
+    public interface IMeleeWeapon
+    {
+        float CurrentCooldown { get; }
+        float MaxCooldown { get; }
+        bool CanMelee { get; }
+    }
+    
+    public class MeleeWeaponBehaviour : MonoBehaviour, IMeleeWeapon
+    {
+        [SerializeField] private MeleeScriptableObject _meleeScriptable;
+        [SerializeField] private Transform _playerTransform;
+
+        [SerializeField] private bool _canAttackMany = true;
+
+        private MeleeWeapon _weapon;
+        private InputAction _meleeActionInputAction;
+
+        public float CurrentCooldown => _weapon?.CurrentCooldown ?? 0f;
+        public float MaxCooldown => _meleeScriptable != null ? _meleeScriptable.AttackCoolDown : 1f;
+        public bool CanMelee => _weapon?.CanMelee ?? false;
+
+        public void Initialize(InputAction meleeAttackAction, IMeleeProperties properties = null)
+        {
+            _meleeActionInputAction = meleeAttackAction;
+            _weapon = new MeleeWeapon(properties ?? _meleeScriptable);
+            BindActions();
+        }
+
+        private void OnMeleeInputAction(InputAction.CallbackContext context)
+        {
+            if (context.performed)
+            {
+                var t = _playerTransform ? _playerTransform : transform;
```

Comment:
NiT; 
`_playerTransform ?? transform`

Reply:
That runs into the whole "doesn't check for unity object lifecycle" issue